### PR TITLE
Add docs for Jupyter NB test suite

### PR DESCRIPTION
**Pull request by etpinard**
follow-up on https://github.com/plotly/plotly.py/pull/549

The pipeline for Jupyter NB tests is anything but trivial, so might as well :books: it!

@theengineear @Kully @cldougl

### CODE REVIEW

File: plotly/tests/test_optional/test_jupyter/test_jupyter.py [hunk 1]
```diff
@@ -12,10 +12,17 @@
...
 PATH_ROOT = path.dirname(__file__)
+PATH_NODE_MODULES = path.join(PATH_ROOT, 'node_modules')
 PATH_FIXTURES = path.join(PATH_ROOT, 'fixtures')
 PATH_JS_TESTS = path.join(PATH_ROOT, 'js_tests')
 
 
+class PlotlyJupyterTestDeps(TestCase):
+
+    def test_node_modules(self):
+        self.assertTrue(path.isdir(PATH_NODE_MODULES))
```
Comment (etpinard) [inline plotly/tests/test_optional/test_jupyter/test_jupyter.py]: @theengineear this test will fail when someone forgets to `npm install` before running these tests. It should make debugging a little easier.

Comment (theengineear) [inline plotly/tests/test_optional/test_jupyter/test_jupyter.py]: Ah, nice! yah, that's a simple first-pass catch for sure.

**Comment by theengineear** [comment]
^^ damn, that image server test *still* failed, rebuilding for now... I need to get to the bottom of that one.

**Comment by theengineear** [comment]
💃 👯 💃

**Pull request merged**